src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModuleFactory.cs

```diff
@@ -38,82 +37,42 @@ namespace Nethermind.JsonRpc.Modules.Eth
+            IBlockchainBridgeFactory blockchainBridgeFactory)
         {
-            _dbProvider = dbProvider ?? throw new ArgumentNullException(nameof(dbProvider));
             _txPool = txPool ?? throw new ArgumentNullException(nameof(txPool));
             _txSender = txSender ?? throw new ArgumentNullException(nameof(txSender));
             _wallet = wallet ?? throw new ArgumentNullException(nameof(wallet));
             _blockTree = blockTree ?? throw new ArgumentNullException(nameof(blockTree));
-            _ethereumEcdsa = ethereumEcdsa ?? throw new ArgumentNullException(nameof(ethereumEcdsa));
-            _receiptFinder = receiptFinder ?? throw new ArgumentNullException(nameof(receiptFinder));
-            _specProvider = specProvider ?? throw new ArgumentNullException(nameof(specProvider));
             _rpcConfig = config ?? throw new ArgumentNullException(nameof(config));
-            _syncConfig = syncConfig ?? throw new ArgumentNullException(nameof(syncConfig));
-            _bloomStorage = bloomStorage ?? throw new ArgumentNullException(nameof(bloomStorage));
             _logManager = logManager ?? throw new ArgumentNullException(nameof(logManager));
-            _isMining = isMining;
-
-            _filterStore = new FilterStore();
-            _filterManager = new FilterManager(_filterStore, blockProcessor, _txPool, _logManager);
+            _stateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
+            _blockchainBridgeFactory = blockchainBridgeFactory ?? throw new ArgumentNullException(nameof(blockchainBridgeFactory));
+            _readOnlyBlockTree = new ReadOnlyBlockTree(_blockTree);
         }
         
         public override IEthModule Create()
         {
-            ReadOnlyBlockTree readOnlyTree = new ReadOnlyBlockTree(_blockTree);
-            IReadOnlyDbProvider readOnlyDbProvider = new ReadOnlyDbProvider(_dbProvider, false);
-            ReadOnlyTxProcessingEnv readOnlyTxProcessingEnv = new ReadOnlyTxProcessingEnv(readOnlyDbProvider, readOnlyTree, _specProvider, _logManager);
-            
-            var blockchainBridge = new BlockchainBridge(
-                readOnlyTxProcessingEnv,
-                _txPool,
-                _receiptFinder,
-                _filterStore,
-                _filterManager,
-                _ethereumEcdsa,
-                _bloomStorage,
-                Timestamper.Default,
-                _logManager,
-                _isMining,
-                _syncConfig.BeamSync && _syncConfig.FastSync,
-                _rpcConfig.FindLogBlockDepthLimit
-                );
-            
-            
             return new EthModule(
                 _rpcConfig,
-                blockchainBridge,
-                readOnlyTxProcessingEnv.BlockTree,
-                readOnlyTxProcessingEnv.StateReader,
+                _blockchainBridgeFactory.CreateBlockchainBridge(),
+                _readOnlyBlockTree,
+                _stateReader,
```

Comment:
I think stateReader might read different state than the one from readonly environment in blockchainbridge

Reply:
this is fine - state reader responds to different questions than processing environments - each RPC call is independent